### PR TITLE
ffmpeg28 formula

### PR DESCRIPTION
**Pull request by boecko**
see #1237

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Looks good to me. @DomT4?

**Comment by boecko** [comment]
Wait .. i'll make it keg-only and i remove the conflict, because some core-formulas depend on ffmpeg.

The keg-only version has its purpose, when you try to make formula for vlc (which i'll do)

Why a vlc-formula?
The VLC-binary from the official-dmg isn't really useful on the cmdline.

cvlc on the cmdline is immensely useful, when you are live transcoding a stream like this.
```bash
/usr/local/bin/cvlc --sout '#transcode{ vcodec=h264, venc=x264{profile=high,vbv-bufsize=1024,bitrate=4500,crf=20}, acodec=mp4a, aenc=ffmpeg{acodec=mp4a,ac=2,ar=44100}, audio-sync, fps=25, deinterlace=0 }:standard{access=http{mime=video/mp4},mux=ts{use-key-frames},dst=:8181/stream.mp4}' --avcodec-hw 1 --avcodec-fast --avcodec-skip-frame 0 --avcodec-skiploopfilter 1 --sout-x264-tune zerolatency --sout-avcodec-hurry-up --sout-transcode-height=576 --sout-x264-preset=veryfast --sout-transcode-high-priority --sout-mono-downmix  --sout-avcodec-strict=-2 <streamurl> 
```

This doesn't work with the official version.

File: ffmpeg28.rb [hunk 1]
```diff
@@ -0,0 +1,193 @@
+class Ffmpeg28 < Formula
+  desc "Play, record, convert, and stream audio and video"
+  homepage "https://ffmpeg.org/"
+  url "https://ffmpeg.org/releases/ffmpeg-2.8.6.tar.bz2"
+  sha256 "40611e329bc354592c6f8f1deb033c31b91f80e91f5707ca4f9afceca78d8e62"
+  head "https://github.com/FFmpeg/FFmpeg.git"
```
Comment (DomT4) [inline ffmpeg28.rb]: Unless it is chained to the specific branch the formula is Versions formulae can't have `head`s.

**Comment by DomT4** [comment]
I'm a little reluctant to carry this many optional dependencies on a Versions formulae because over time the Versions formulae will stay where it is in terms of versioning and the dependencies will race on unchained, which seems likely to lead to significant trouble down the line. I'll take it as-is, but that's worth thinking about.

> when you try to make formula for vlc (which i'll do)

FWIW in advance you can't have mandatory cross-tap dependencies, if that was something you're considering submitting to `homebrew/core`.

> Looks good to me. @DomT4?

Took me a second to recognise your account here with the new avatar 😄. Big ❤️ for dog-featuring profile pictures.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Will take this as-is, for now. We'll see how it goes in terms of long-term support, but it has a valid place in Versions.

Merged in https://github.com/Homebrew/homebrew-versions/commit/2920a11f66ee81460c56eb2ec92fd6d58e940641. Thank you for your contribution to Homebrew @boecko; we appreciate it! :smiley_cat:

**Pull request closed without merging**